### PR TITLE
add an indexer to SymbolicShape

### PR DESCRIPTION
**Pull request by Krovatkin**
A convenience indexer into `SymbolicShape`s

### CODE REVIEW

File: aten/src/ATen/core/jit_type.h [hunk 1]
```diff
@@ -548,7 +555,7 @@ struct VaryingShape {
     return dims_ == other.dims_;
   }
 
-  const c10::optional<T>& operator[](int i) const {
+  const c10::optional<T> &operator[](size_t i) const {
```
Comment (ZolotukhinM) [inline aten/src/ATen/core/jit_type.h:558]: Nit: this seems to include unnecessary formatting change which I assume would break clang-format?

**Review by ZolotukhinM — APPROVED** [review]
Looks good!

File: aten/src/ATen/core/jit_type.h (hunk 1 above)
Comment (Krovatkin) [reply aten/src/ATen/core/jit_type.h:558]: this **was** clang-format :-) But let's see if it breaks our lint jobs.
